PostPairsTrading.md

```diff
@@ -0,0 +1,323 @@
+
+The implementation of these functions is based on the following formulas:
+
+$\beta = \frac{{(n \cdot \sum(x \cdot y)) - (\sum x \cdot \sum y)}}{{(n \cdot \sum(x^2)) - (\sum x)^2}}$
+
+$\alpha = \text{Mean}(y) - \beta \cdot \text{Mean}(x)$
+
+---
+
+We've already calculated the alpha and beta using the logarithmic values of our historical data (since we don't have prior knowledge of the real-time price values for priceX and priceY). Now, all that remains is to apply the previous formula to derive our spreads:
+
+---
+
+```q
+spreads: historical_data_priceY - ((historical_data_priceX*beta)+alpha) // = -0.1493929 0.0451223 -0.08835117 0.0451223 0.1579725
+```
+
+> 🖥️ In Q/kdb+, operand priority is strictly from right to left, without any precedence rules except those involving parentheses. Therefore, it's crucial to exercise caution when writing Q code to ensure accurate results.
+
+---
+
+And this precisely meets our objective—a **comprehensive method for representing relative changes between both assets**. As we can deduce, our mean is now 0 because our assets are normalized, cointegrated and on the same scale. Therefore, ideally, the differential between their prices should be 0. Consequently, when our spread is below 0, we infer that asset X is overpriced, whereas if it's above 0, then asset Y is overpriced.
+
+![Spreads](https://github.com/hablapps/pairstrading/blob/5-Post/Spreads.gif?raw=true)
+
+
+## Two steps forward, one step back.
+
+Before proceeding to plot the NASDAQ100-SP500 spreads, we need to first plan our algorithm. In this post, we intend to create **a real-time scenario** for Pairs trading, so careful planning is essential for our sake.
+
+Decomposing our steps, let's start from the very beginning:
+
+1. Data management part
+
+        1.1. Read data
+
+        1.2. Filter data
+
+2. Outside the loop
+
+        2.1. Linear regression
+
+        2.2. Initialize values 
+
+3. Inside the loop
+
+        3.1. Calculate spreads
+
+        3.2. Write on buffers
+
+
+As previously mentioned, historical data is crucial for generating accurate spreads. We need to calculate each spread in real-time **using precomputed alpha and beta values** derived from both prices. Therefore, once we have obtained our historical values, we can proceed with calculating our linear regression.
+
+---
+
+```q
+// Fix data and take log(prices) -> Simulated data
+priceX: 0!1_(update delta:0f^deltas dateTime from 
+        distinct select distinct dateTime, log bid, log ask from
+        (update dateTime:"P"$@[;19;:;"."] each dateTime from 
+        tab1) where not null bid);
+
+priceY: 0!1_(update delta:0f^deltas dateTime from 
+        distinct select distinct dateTime, log bid, log ask from 
+        (update dateTime:"P"$@[;19;:;"."] each dateTime from 
+        tab2) where not null bid);

```

Comment:
These definitions are essentially the same, up to the table they read from. Modularise it. By the way, what is `tab1` and `tab2`? What is going on with the hardcoded `19` value?